src/pages/projects/_components/ProjectForm.tsx

```diff
@@ -1,54 +1,145 @@
+import { useProject } from "../../../hooks/useProject";
 
 type Props = {
-  project?: {
-    id: string;
-    name: string;
-  }
+  projectId?: string
+
 };
 
-const ProjectForm = (props: Props) => {
-  const { project } = props;
+const ProjectForm = ({ projectId }: Props) => {
+  let navigate = useNavigate();
+  const { data, isLoading, error, upsertProject } = useProject(projectId);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitSuccessful, isDirty },
+    reset
+  } = useForm<{
+    id?: string;
+    name: string;
+    color: string;
+  }>({
+    defaultValues: {
+      name: data?.name,
+      color: data?.color
+    },
+  });
+
+  const onSubmit = async (values: { id?: string; name: string; color: string; }) => {
+    await upsertProject(values);
+  };
+
+  useEffect(() => {
+    if (data)
+      reset(data);
+  }, [data]);
+
+  const [dialog, setDialog] = useState({
+    message: "",
+    openModal: false,
+    title: ""
+  });
+
+  const handleDialog = (message: string, openModal: boolean, title: string) => {
+    setDialog({
+      message,
+      openModal,
+      title
+    });
+  };
+
+  const handleCancel = () => {
+    if (isDirty)
+      handleDialog("You will lost the data", true, "Cancel confirmation");
+    else {
+      return navigate("/projects");
+    }
+  };
+
+  const areSureCancel = (choose: boolean) => {
+    if (choose) {
+      return navigate("/projects");
+    } else {
+      handleDialog("", false, "");
+    }
+  };
 
   return (
     <div>
-      <h1 className="text-3xl font-bold underline text-blue-500">
-        {project ? "Edit Project" : "Create Project"}
-      </h1>
-      <div className="mt-4">
-        <form>
-          <div className="mb-4">
-            <label className="block text-gray-700 text-sm font-bold mb-2">
-              Name
-            </label>
-            <input
-              className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
-
-              type="text"
-              placeholder="Project name"
-              defaultValue={project?.name}
+      <div className="container max-w-screen-sm">
+        <h1 className="text-3xl font-bold mb-10">
+          {data ? "Edit Project" : "Create Project"}
+        </h1>
+        <div className="mt-4">
+          <form onSubmit={handleSubmit(onSubmit)}>
+            {data != null && <><div className="mt-4">
+              <label className="block text-gray-700 text-sm font-bold mb-2">
+                id
+              </label>
+            </div><input disabled={true} defaultValue={data?.id} className="input input-bordered w-full "
+              /></>}
+            <div className="mt-4">
+              <label className="block text-gray-700 text-sm font-bold mb-2">
+                Name
+              </label>
+            </div>
+            <input placeholder="billing system zgt client" className="input input-bordered w-full "
```

Comment:
empty space at end of className property

Reply:
fixed